File: week3/data_transformation.py
# Standardization example
from sklearn.preprocessing import StandardScaler

# Create a scaler object
scaler = StandardScaler()

# Fit and transform the data
scaled_data = scaler.fit_transform(numerical_data)

# One-hot encoding for categorical variables
import pandas as pd

# Create dummy variables
one_hot_encoded = pd.get_dummies(categorical_column)

File: week3/data_transformation.r
# Standardization example
# Using base R
scaled_data <- scale(numerical_data)

# Using dplyr
library(dplyr)
scaled_data <- numerical_data %>%
  mutate_all(~(.-mean(.))/sd(.))

# One-hot encoding
library(fastDummies)
one_hot_encoded <- dummy_cols(data, select_columns = "categorical_column")
